google-cloud-firestore/src/main/java/com/google/cloud/firestore/telemetry/EnabledTraceUtil.java

```diff
@@ -84,6 +85,11 @@ public ManagedChannelBuilder apply(ManagedChannelBuilder managedChannelBuilder)
   @Override
   @Nullable
   public ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder> getChannelConfigurator() {
+    // Note: using `==` rather than `.equals` since OpenTelemetry has only 1 static instance of
+    // `TracerProvider.noop`.
+    if (openTelemetry.getTracerProvider() == TracerProvider.noop()) {
```

Comment:
IIRC, an OTEL instance can register multiple MetricsProvider, even though it did not work as expected when I try on a test app. 
I expect TracerProvider would be the same. https://opentelemetry.io/docs/specs/otel/trace/api/#tracerprovider
`some applications may want to or have to use multiple TracerProvider instances`